pkg/wg_timeout/wg_timeout.go

```diff
@@ -0,0 +1,25 @@
+package wg_timeout
+
+import (
+	"sync"
+	"time"
+)
+
+func WaitGroupTimeout(wg *sync.WaitGroup, timeout time.Duration) bool {
```

Comment:
I would take a `zap.Logger` and a message here.  Print the message at `Warn` level each time around the loop, with the total elapsed duration as a `zap` field. No return value - make this like `Wait`

I would not return on  timer ticks; that will change the behavior of the system. Just log the wait. 

Style suggestion: take `c` as an argument to the lambda, restricting its directionality, and renaming it.